bootstrap/templates/kubernetes/talos/talconfig.yaml.j2

```diff
@@ -0,0 +1,194 @@
+#% if bootstrap_distribution == 'talos' %#
+# yaml-language-server: $schema=https://raw.githubusercontent.com/budimanjojo/talhelper/master/pkg/config/schemas/talconfig.json
+---
+# renovate: datasource=github-releases depName=siderolabs/talos
+talosVersion: v1.6.1
+# renovate: datasource=github-releases depName=kubernetes/kubernetes extractVersion=^v(?<version>.*)$
+kubernetesVersion: 1.29.1
+
+clusterName: &cluster home-kubernetes
+
+#% if bootstrap_kube_api_hostname | default('') %#
+endpoint: https://#{ bootstrap_kube_api_hostname }#:6443
+#% else %#
+endpoint: https://#{ bootstrap_kube_api_addr }#:6443
+#% endif %#
+
+clusterPodNets:
+  - "#{ bootstrap_cluster_cidr.split(',')[0] }#"
+clusterSvcNets:
+  - "#{ bootstrap_service_cidr.split(',')[0] }#"
+
+additionalApiServerCertSans: &sans
+  - "#{ bootstrap_kube_api_addr }#"
+  - 127.0.0.1 #KubePrism
+  #% if bootstrap_kube_api_hostname | default('') %#
+  - "#{ bootstrap_kube_api_hostname }#"
+  #% endif %#
+
+additionalMachineCertSans: *sans
+
+cniConfig:
+  name: none
+
+nodes:
+  #% for item in bootstrap_nodes.master %#
+  - hostname: "#{ item.name }#"
+    ipAddress: &ipAddress "#{ item.address }#"
+    installDiskSelector:
+      serial: "#{ item.diskSerial }#"
+    controlPlane: true
+    networkInterfaces:
+      - interface: eth0
+        dhcp: false
+        mtu: 1500
+        addresses:
+          - *ipAddress
+        routes:
+          - network: 0.0.0.0/0
+            gateway: "#{ bootstrap_node_cidr | nthhost(1) }#"
+        vip:
+          ip: "#{ bootstrap_kube_api_addr }#"
```

Comment:
I would wrap this into the patches section.  Since it is applied to all nodes it can be there instead of explicitly configured for each node. you can also cut out the addresses, the ipAddress property sets that property in the machine config.

Reply:
The `vip.ip` in that block section only applies to the master nodes right?

Reply:
If so, you cannot merge arrays with yaml. Also looping in jinja makes it hard to reduce the LoC using yaml anchors here.

Reply:
I removed the `addresses` part, we can always re-iterate on patches in the future but this works for now.